File: TheaterSuggestions/CSharp/SeatsSuggestions/Seat.cs
namespace SeatsSuggestions
{
    public class Seat(string rowName, uint number, PricingCategory pricingCategory, SeatAvailability seatAvailability)
    {
        public string RowName { get; } = rowName;
        public uint Number { get; } = number;
        public PricingCategory PricingCategory { get; } = pricingCategory;
        private SeatAvailability SeatAvailability { get; set; } = seatAvailability;

        public bool IsAvailable()
        {
            return SeatAvailability == SeatAvailability.Available;
        }

        public override string ToString()
        {
            return $"{RowName}{Number}";
        }

        public bool MatchCategory(PricingCategory pricingCategory)
        {
            if (pricingCategory == PricingCategory.Mixed)
            {
                return true;
            }

            return PricingCategory == pricingCategory;
        }

        public void Allocate()
        {
            if (SeatAvailability == SeatAvailability.Available)
            {
                SeatAvailability = SeatAvailability.Allocated;
            }
        }
    }
}

File: TheaterSuggestions/Java/SeatsSuggestions/src/main/java/com/octo/technology/SeatsSuggestions/Seat.java
package com.octo.technology.SeatsSuggestions;

public class Seat {

    private final String rowName;
    private final int number;
    private final PricingCategory pricingCategory;
    private SeatAvailability seatAvailability;

    public Seat(String rowName, int number, PricingCategory pricingCategory, SeatAvailability seatAvailability) {
        this.rowName = rowName;
        this.number = number;
        this.pricingCategory = pricingCategory;
        this.seatAvailability = seatAvailability;
    }

    public boolean isAvailable() {
        return seatAvailability == SeatAvailability.Available;
    }

    public boolean matchCategory(PricingCategory pricingCategory) {
        if (pricingCategory == PricingCategory.Mixed) {
            return true;
        }

        return this.pricingCategory == pricingCategory;
    }

    public void allocate() {
        if (seatAvailability == SeatAvailability.Available) {
            seatAvailability = SeatAvailability.Allocated;
        }
    }

    @Override
    public String toString() {
        return rowName + number;
    }
}
